Engine/src/Engine/ResourceManager.cpp

```diff
@@ -262,6 +222,105 @@ namespace Engine
 	
 	}
 
+	//Based on the provided spritesheet filename, return the vector of GLuints for the sprites stored within a map, or load it from the system.
+	ResourceManager::SpriteSheet ResourceManager::getSpritesheet(const std::string& name, float spriteWidth, float spriteHeight)
+	{
+		//Initilaze the return value, and create a map iterator to check if the data is already loaded.
+		SpriteSheet data {0,0,0,0,0,0,0,0,0,NULL};
+
+		auto it = m_spritesheetsTex.find(name);
+
+		//If loaded, return the spritesheet. Else, load and store the image, then return it.
+		if (it != m_spritesheetsTex.end())
+		{
+			//Get value based on map key "name"
+			SpriteSheet data = m_spritesheetsTex.find(name)->second;
+
+			GE_CORE_WARN("[ResourceManager] Previously loaded spritesheet " + name + " found.");
+			return data;
+		}
+
+		ImageData img = readImageData(name);
+
+		//If image data isn't nullptr, store and return data.
+		if (img.image != nullptr)
+		{
+			ImageData spriteSheet = getTexture(name);
+
+			if (spriteWidth == 0 || spriteHeight == 0)
+			{
+				spriteWidth = spriteSheet.width / 3;
```

Comment:
assuming the number of frames in the spritesheet might prove problematic in the future. I would suggest either keeping a variable in the spritesheet struct, or passing it in as a function param.

Reply:
if We're assuming 3 then that can be the default value of either variable/param

Reply:
So with this, typically the frames would be calculated if the function is passed the dimensions of the sprite. 

That if statement is to catch any cases where the dimensions are 0, in which case we would assume our default spritesheet dimensions (the size of the sheet split into 3 columns/rows). 

In terms of defaulting the sprite width/height, that might be hard because that would change based on the size of the spritesheet (and we don't obtain that data until here). 

I'm open to suggestions though

Reply:
I meant defaulting the "3" in the code snippet. Otherwise, it's a magic variable, and it's hard to understand unless we know of the default sprite sheet we would expect